sway/criteria.c

```diff
@@ -137,21 +144,35 @@ static char *crit_tokens(int *argc, char ***buf, const char * const criteria_str
 					"Found quoted value where it was not expected");
 			} else if (!valp) { // value starts here
 				valp = head + 1;
+				quoted = true;
 			} else {
 				// value ends here
 				argv[(*argc)++] = valp;
 				*head = '\0';
 				valp = NULL;
 				namep = head + 1;
 			}
-		} else if (*argc % 2 == 0 && !valp && *head != ' ') {
-			// We're expecting a quoted value, haven't found one yet, and this
-			// is not an empty space.
-			return strdup("Unable to parse criteria: "
-				"Names must be unquoted, values must be quoted");
+		} else if (*argc % 2 == 0 && *head != ' ') {
+			// parse unquoted values
+			if (!valp) {
+				quoted = false;
+				valp = head;  // value starts here
+			}
+		} else if (valp && !quoted && *head == ' ') {
+			// value ends here
+			argv[(*argc)++] = valp;
```

Comment:
Can you split this on to two lines? I don't like the dereference-increment operation going on here.

Reply:
Yes, I agree. I picked the existing style of the function.